### PR TITLE
Do not remove usedGKs when all resources are deleted

### LINKED ISSUE #929: kapp flip flops betweek gk scoping and non scoping while using `--dangerous-allow-empty-list-of-resources`
**What steps did you take:**
I have a Gitops system where we deploy resources from files using kapp. We just use `kapp deploy` to deploy and delete resources, and hence we have to use the `--dangerous-allow-empty-list-of-resources` flag when all files are removed.
```bash
$ kapp deploy -a test -f examples/simple-app/config-1.yml
.
.
.
// Deployment and Service Created
$ kapp deploy -a test -f empty-dir --dangerous-allow-empty-list-of-resources
.
.
.
// Deployment and Service deleted. 
```

**What happened:**
After the above two commands are run, kapp removes the `usedGKs` list from the meta ConfigMap. Due to this, the next time kapp deploy is run, kapp doesn't use GK scoping because it thinks that this is an old app which didn't have gk scoping enabled.

**What did you expect:**
I expected kapp to continue scope the listing to the GKs from the resources I am trying to deploy.

**Anything else you would like to add:**

**Environment:**

- kapp version (use `kapp --version`):
- OS (e.g. from `/etc/os-release`):
- Kubernetes version (use `kubectl version`)

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by praveenrewar**
#### What this PR does / why we need it:
Do not remove usedGKs when all resources are deleted.

#### Which issue(s) this PR fixes:

Fixes #929 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by prembhaskal** [comment]
LGTM

File: pkg/kapp/app/recorded_app.go [hunk 1]
```diff
@@ -105,8 +105,13 @@ func (a *RecordedApp) UpdateUsedGVsAndGKs(gvs []schema.GroupVersion, gks []schem
...
 	}
 
+	sort.Slice(uniqGVs, func(i int, j int) bool {
+		return uniqGVs[i].Group+uniqGVs[i].Version < uniqGVs[j].Group+uniqGVs[j].Version
+	})
+
 	gksByGK := map[schema.GroupKind]struct{}{}
-	var uniqGKs []schema.GroupKind
+	// Initialize with an empty slice so that we do not remove `usedGKs` from meta configmap
+	uniqGKs := []schema.GroupKind{}
```
Comment (100mik) [inline pkg/kapp/app/recorded_app.go:114]: Do we want to make the same change for GVs as well?

Comment (praveenrewar) [reply pkg/kapp/app/recorded_app.go:114]: We don't have a similar check on GVs, so it won't really be helpful.
Fir GKs, if it's nil, then we assume that it's an old app and we don't scope list calls to used GKs.

Comment (100mik) [reply pkg/kapp/app/recorded_app.go:114]: Gotcha, makes sense

**Review by 100mik — APPROVED** [review]
LGTM!

**Pull request merged**